x/nftmarket/keeper/keeper.go

```diff
@@ -5,41 +5,57 @@ import (
 
 	"github.com/tendermint/tendermint/libs/log"
 
-	"github.com/UnUniFi/chain/x/nftmarket/types"
+	"github.com/cosmos/cosmos-sdk/client"
 	"github.com/cosmos/cosmos-sdk/codec"
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
+	"github.com/UnUniFi/chain/x/nftmarket/types"
 )
 
 type Keeper struct {
 	cdc           codec.Codec
+	txCfg         client.TxConfig
 	storeKey      storetypes.StoreKey
 	memKey        storetypes.StoreKey
 	paramSpace    paramtypes.Subspace
 	accountKeeper types.AccountKeeper
 	bankKeeper    types.BankKeeper
 	nftKeeper     types.NftKeeper
+	hooks         types.NftmarketHooks
 }
 
-func NewKeeper(cdc codec.Codec, storeKey, memKey storetypes.StoreKey,
-	paramSpace paramtypes.Subspace, accountKeeper types.AccountKeeper,
-	bankKeeper types.BankKeeper, nftKeeper types.NftKeeper) Keeper {
+func NewKeeper(cdc codec.Codec, txCfg client.TxConfig, storeKey,
+	memKey storetypes.StoreKey, paramSpace paramtypes.Subspace,
+	accountKeeper types.AccountKeeper, bankKeeper types.BankKeeper,
+	nftKeeper types.NftKeeper) Keeper {
 	if !paramSpace.HasKeyTable() {
 		paramSpace = paramSpace.WithKeyTable(types.ParamKeyTable())
 	}
 
 	return Keeper{
 		cdc:           cdc,
+		txCfg:         txCfg,
 		storeKey:      storeKey,
 		memKey:        memKey,
 		paramSpace:    paramSpace,
 		accountKeeper: accountKeeper,
 		bankKeeper:    bankKeeper,
 		nftKeeper:     nftKeeper,
+		hooks:         nil,
 	}
 }
 
 func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
+
+// SetHooks sets the nftmarket keeper hooks
+func (k *Keeper) SetHooks(hooks types.NftmarketHooks) *Keeper {
+	if k.hooks != nil {
```

Comment:
Is that not the wrong conditional expression?

Reply:
no, because it's supposed to be called only when to create app struct to avoid conflicts of hook methods registration, i guess.
this is correct.